### PR TITLE
feat(migration): Add like question action handler in the UI

### LINKED ISSUE #35: Add vote question functionality
Add the vote question functionality in the list of questions

### PR DESCRIPTION
**Pull request by AldoDominguezWizeline**
#### What does this PR do?
It adds the ability to like a question both in the question list index page and in the question detail page, debouncing multiple clicks in rapid succession using the useState hook available for Remix.

https://user-images.githubusercontent.com/104774638/196557306-842e27d3-4354-439a-80f7-bdc0c0f96c9b.mov

#### Where should the reviewer start?
- app/components/ListQuestions/ListQuestions.jsx
- app/routes/index.jsx
- app/components/QuestionDetail/QuestionDetail.jsx
- app/routes/questions/$questionId.jsx

#### How should this be manually tested?
1. Apply some filtering criteria in the list of questions that has some results
2. Vote one of the filtered questions
3. The search parameters should be kept after submitting the like action
4. Open the details of a question and verify that we can like the question there as well, with a 'not-allowed' cursor when hovering over the like button when there's a like submission being processed.

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #35

### CODE REVIEW

File: app/components/CounterButton/CounterButton.Styled.jsx [hunk 1]
```diff
@@ -9,13 +9,22 @@ const handleColorType = (props) => {
   return '#31425a';
 };
 
+const setCursorType = (props) => {
+  if (props.processingFormSubmission) {
+    return 'not-allowed;';
+  }
+  else {
```
Comment (Hoosep) [inline app/components/CounterButton/CounterButton.Styled.jsx:16]: This else is not necessary. Just return your string.

File: app/components/CounterButton/CounterButton.Styled.jsx [hunk 2]
```diff
@@ -9,13 +9,22 @@ const handleColorType = (props) => {
   return '#31425a';
 };
 
+const setCursorType = (props) => {
+  if (props.processingFormSubmission) {
+    return 'not-allowed;';
```
Comment (Hoosep) [inline app/components/CounterButton/CounterButton.Styled.jsx:14]: I think you don't need the semicolon. I saw where you call this function and it already has the semicolon.

File: app/components/CounterButton/CounterButton.Styled.jsx [hunk 3]
```diff
@@ -9,13 +9,22 @@ const handleColorType = (props) => {
...
+};
+
 export const ContainerCounterButton = styled.button`
   align-items: center;
   background-color: transparent;
   border: none;
   border-radius: 3px;
   color: ${props => handleColorType(props)};
-  cursor: pointer;
+  cursor: ${props => setCursorType(props)};
```
Comment (Hoosep) [inline app/components/CounterButton/CounterButton.Styled.jsx:27]: This calling.

File: app/components/ListQuestions/ListQuestions.jsx [hunk 4]
```diff
@@ -77,6 +80,21 @@ const ListQuestions = ({
 
 
   const renderQuestionsList = () => {
+    const onLikeButtonClick = async (questionId) => {
```
Comment (Hoosep) [inline app/components/ListQuestions/ListQuestions.jsx:83]: Async is required? I don't see any await inside function

File: app/components/QuestionDetail/QuestionDetail.jsx [hunk 5]
```diff
@@ -46,6 +50,17 @@ function QuestionDetails(props) {
   };
 
   const renderQuestionButtons = () => {
+    const onLikeButtonClick = async () => {
```
Comment (Hoosep) [inline app/components/QuestionDetail/QuestionDetail.jsx:53]: Same thing. Async is required?

**Review by Hoosep — CHANGES_REQUESTED** [review]

**Comment by Hoosep** [comment]
I left you some comments. Any doubt, please reach me!:)

File: app/components/CounterButton/CounterButton.Styled.jsx (hunk 1 above)
Comment (AldoDominguezWizeline) [reply app/components/CounterButton/CounterButton.Styled.jsx]: You are totally right, sorry about that, I've removed this redundant else clause, thank you!

File: app/components/CounterButton/CounterButton.Styled.jsx (hunk 2 above)
Comment (AldoDominguezWizeline) [reply app/components/CounterButton/CounterButton.Styled.jsx]: You are right about this unnecessary semicolon, I've removed it as well, thank you!

File: app/components/CounterButton/CounterButton.Styled.jsx (hunk 3 above)
Comment (AldoDominguezWizeline) [reply app/components/CounterButton/CounterButton.Styled.jsx:25]: Exactly, the semicolon is already there, no need for it in the setCursorType function, thank you!

File: app/components/ListQuestions/ListQuestions.jsx (hunk 4 above)
Comment (AldoDominguezWizeline) [reply app/components/ListQuestions/ListQuestions.jsx]: You are totally right about both functions not being asynchronous, I had forgotten to remove the keyword, I've removed it now, thank you!

File: app/components/QuestionDetail/QuestionDetail.jsx (hunk 5 above)
Comment (AldoDominguezWizeline) [reply app/components/QuestionDetail/QuestionDetail.jsx]: Removed unnecessary async here as well

**Comment by AldoDominguezWizeline** [comment]
@Hoosep Thank you very much for the observations, I've addressed the comments, thanks again!

**Review by Hoosep — APPROVED** [review]
LGTM! 🎉✨

**Review by GeorginaFraire — APPROVED** [review]
LGTM 🎉

**Pull request merged**